test/errors-spec.js

```diff
@@ -91,84 +90,85 @@ describe("Parser:", function() {
     });
 
     describe("#parseAtom", function() {
-        // TODO: The positions in the following error messages appear to be
-        // off by one, i.e. they should be one character later.
         it("rejects \\limits without operator", function() {
             expect("\\alpha\\limits\\omega").toFailWithParseError(
                    "Limit controls must follow a math operator" +
-                   " at position 6: \\alpha̲\\limits\\omega");
+                   " at position 7: \\alpha\\̲l̲i̲m̲i̲t̲s̲\\omega");
```

Comment:
Well, the old `ParseError` implementation did take the first `position` characters, and then add a combining underscore to that. Which means that for highlighting it assumed its positions to be one-based: you'd have to take one character if you wanted to add an underscore to the first character of the string. Probably due to lack of testing, many error messages returned the zero-based position of the problem, though, since zero-based indexing is natural in JavaScript. I left the one-based indexing in the error message for now, but tried to fix the internal positions to actually highlight where the problem is, not off by one. If you look at the deleted Parser code, you will notice that *some* of the `ParseError` constructor calls have a `+ 1` in their position arguments, while others do not.